To: vim_dev@googlegroups.com
Subject: Patch 7.3.085
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.085 (after 7.3.083)
Problem:    Inconsistency with preproc symbols.  void * computation.
Solution:   Include vimio.h from vim.h.  Add type cast.
Files:      src/eval.c, src/ex_cmds.c, src/ex_cmds2.c, src/fileio.c,
            src/if_cscope.c, src/if_sniff.c, src/main.c, src/memfile.c,
            src/memline.c, src/netbeans.c, src/os_msdos.c, src/os_mswin.c,
            src/os_win16.c, src/os_win32.c, src/spell.c, src/tag.c,
            src/undo.c, src/vim.h


*** ../vim-7.3.084/src/eval.c	2010-12-02 21:44:35.000000000 +0100
--- src/eval.c	2010-12-17 17:45:37.000000000 +0100
***************
*** 10,18 ****
  /*
   * eval.c: Expression evaluation.
   */
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for mch_open(), must be before vim.h */
- #endif
  
  #include "vim.h"
  
--- 10,15 ----
*** ../vim-7.3.084/src/ex_cmds.c	2010-11-24 17:59:27.000000000 +0100
--- src/ex_cmds.c	2010-12-17 17:46:09.000000000 +0100
***************
*** 11,20 ****
   * ex_cmds.c: some functions for command line commands
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for mch_open(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  #include "version.h"
  
--- 11,16 ----
*** ../vim-7.3.084/src/ex_cmds2.c	2010-09-21 16:56:29.000000000 +0200
--- src/ex_cmds2.c	2010-12-17 17:46:26.000000000 +0100
***************
*** 11,20 ****
   * ex_cmds2.c: some more functions for command line commands
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for mch_open(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  #include "version.h"
  
--- 11,16 ----
*** ../vim-7.3.084/src/fileio.c	2010-12-17 16:27:09.000000000 +0100
--- src/fileio.c	2010-12-17 17:52:42.000000000 +0100
***************
*** 11,24 ****
   * fileio.c: read from and write to a file
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for lseek(), must be before vim.h */
- #endif
- 
- #if defined __EMX__
- # include "vimio.h"	/* for mktemp(), CJW 1997-12-03 */
- #endif
- 
  #include "vim.h"
  
  #if defined(__TANDEM) || defined(__MINT__)
--- 11,16 ----
***************
*** 10336,10342 ****
       * by a signal. */
      while (ret < (long)bufsize)
      {
! 	wlen = vim_write(fd, buf + ret, bufsize - ret);
  	if (wlen < 0)
  	{
  	    if (errno != EINTR)
--- 10328,10334 ----
       * by a signal. */
      while (ret < (long)bufsize)
      {
! 	wlen = vim_write(fd, (char *)buf + ret, bufsize - ret);
  	if (wlen < 0)
  	{
  	    if (errno != EINTR)
*** ../vim-7.3.084/src/if_cscope.c	2010-09-21 16:56:29.000000000 +0200
--- src/if_cscope.c	2010-12-17 17:47:02.000000000 +0100
***************
*** 20,28 ****
  #include <sys/stat.h>
  #if defined(UNIX)
  # include <sys/wait.h>
- #else
-     /* not UNIX, must be WIN32 */
- # include "vimio.h"
  #endif
  #include "if_cscope.h"
  
--- 20,25 ----
*** ../vim-7.3.084/src/if_sniff.c	2010-08-15 21:57:25.000000000 +0200
--- src/if_sniff.c	2010-12-17 17:47:20.000000000 +0100
***************
*** 9,15 ****
  
  #ifdef WIN32
  # include <stdio.h>
- # include "vimio.h"
  # include <process.h>
  # include <string.h>
  # include <assert.h>
--- 9,14 ----
*** ../vim-7.3.084/src/main.c	2010-11-16 16:25:46.000000000 +0100
--- src/main.c	2010-12-17 17:47:41.000000000 +0100
***************
*** 7,16 ****
   * See README.txt for an overview of the Vim source code.
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"		/* for close() and dup() */
- #endif
- 
  #define EXTERN
  #include "vim.h"
  
--- 7,12 ----
*** ../vim-7.3.084/src/memfile.c	2010-12-17 16:27:10.000000000 +0100
--- src/memfile.c	2010-12-17 17:47:54.000000000 +0100
***************
*** 32,41 ****
   * file is opened.
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for lseek(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  /*
--- 32,37 ----
*** ../vim-7.3.084/src/memline.c	2010-12-17 16:27:10.000000000 +0100
--- src/memline.c	2010-12-17 17:48:06.000000000 +0100
***************
*** 42,51 ****
   *  mf_get().
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for mch_open(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  #ifndef UNIX		/* it's in os_unix.h for Unix */
--- 42,47 ----
*** ../vim-7.3.084/src/netbeans.c	2010-12-17 12:19:14.000000000 +0100
--- src/netbeans.c	2010-12-17 17:48:19.000000000 +0100
***************
*** 16,25 ****
   * See ":help netbeans-protocol" for explanation.
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for mch_open(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  #if defined(FEAT_NETBEANS_INTG) || defined(PROTO)
--- 16,21 ----
*** ../vim-7.3.084/src/os_msdos.c	2010-08-15 21:57:25.000000000 +0200
--- src/os_msdos.c	2010-12-17 17:48:40.000000000 +0100
***************
*** 21,27 ****
   * Some functions are also used for Win16 (MS-Windows 3.1).
   */
  
- #include "vimio.h"
  #include "vim.h"
  
  #include <conio.h>
--- 21,26 ----
*** ../vim-7.3.084/src/os_mswin.c	2010-10-23 14:02:48.000000000 +0200
--- src/os_mswin.c	2010-12-17 17:48:51.000000000 +0100
***************
*** 22,28 ****
  # endif
  #endif
  
- #include "vimio.h"
  #include "vim.h"
  
  #ifdef WIN16
--- 22,27 ----
*** ../vim-7.3.084/src/os_win16.c	2010-08-15 21:57:32.000000000 +0200
--- src/os_win16.c	2010-12-17 17:49:02.000000000 +0100
***************
*** 20,26 ****
  # pragma warn -obs
  #endif
  
- #include "vimio.h"
  #include "vim.h"
  
  #include <dos.h>
--- 20,25 ----
*** ../vim-7.3.084/src/os_win32.c	2010-11-24 12:35:14.000000000 +0100
--- src/os_win32.c	2010-12-17 17:49:11.000000000 +0100
***************
*** 20,26 ****
   * Roger Knobbe <rogerk@wonderware.com> did the initial port of Vim 3.0.
   */
  
- #include "vimio.h"
  #include "vim.h"
  
  #ifdef FEAT_MZSCHEME
--- 20,25 ----
*** ../vim-7.3.084/src/spell.c	2010-12-08 17:09:27.000000000 +0100
--- src/spell.c	2010-12-17 17:49:24.000000000 +0100
***************
*** 303,312 ****
   *			    few bytes as possible, see offset2bytes())
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for lseek(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  #if defined(FEAT_SPELL) || defined(PROTO)
--- 303,308 ----
*** ../vim-7.3.084/src/tag.c	2010-09-21 16:56:29.000000000 +0200
--- src/tag.c	2010-12-17 17:49:35.000000000 +0100
***************
*** 11,20 ****
   * Code to handle tags and the tag stack
   */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for lseek(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  /*
--- 11,16 ----
*** ../vim-7.3.084/src/undo.c	2010-12-17 16:27:10.000000000 +0100
--- src/undo.c	2010-12-17 17:49:59.000000000 +0100
***************
*** 81,90 ****
  #define UH_MAGIC 0x18dade	/* value for uh_magic when in use */
  #define UE_MAGIC 0xabc123	/* value for ue_magic when in use */
  
- #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64)
- # include "vimio.h"	/* for vim_read(), must be before vim.h */
- #endif
- 
  #include "vim.h"
  
  static void u_unch_branch __ARGS((u_header_T *uhp));
--- 81,86 ----
*** ../vim-7.3.084/src/vim.h	2010-12-17 16:27:10.000000000 +0100
--- src/vim.h	2010-12-17 17:51:43.000000000 +0100
***************
*** 27,32 ****
--- 27,37 ----
  # endif
  #endif
  
+ #if defined(MSDOS) || defined(WIN16) || defined(WIN32) || defined(_WIN64) \
+ 	|| defined(__EMX__)
+ # include "vimio.h"
+ #endif
+ 
  /* ============ the header file puzzle (ca. 50-100 pieces) ========= */
  
  #ifdef HAVE_CONFIG_H	/* GNU autoconf (or something else) was here */
*** ../vim-7.3.084/src/version.c	2010-12-17 17:35:05.000000000 +0100
--- src/version.c	2010-12-17 17:55:39.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     85,
  /**/

-- 
How To Keep A Healthy Level Of Insanity:
14. Put mosquito netting around your work area. Play a tape of jungle
    sounds all day.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
